Tests/RegressionTests/PathExtractionSupport.swift

```diff
@@ -0,0 +1,245 @@
+    case .addLineToPoint:
+      let to = points.pointee
+      return .lineTo(to)
+    case .addQuadCurveToPoint:
+      assertionFailure("addQuadCurveToPoint is not supported")
+      return .empty
+    case .addCurveToPoint:
+      let control1 = points.pointee
+      let control2 = points.advanced(by: 1).pointee
+      let to = points.advanced(by: 2).pointee
+      return .curveTo(control1, control2, to)
+    case .closeSubpath:
+      return .closePath
+    @unknown default:
+      assertionFailure("unknown cgpath element")
+      return .empty
+    }
+  }
+}
+
+extension CGPoint {
+  fileprivate var svgDescription: String {
+    "\(x),\(y)"
+  }
+}
+
+private func svgString(
+  pathDescription: String,
+  id: String
+) -> String {
+  """
+  <?xml version="1.0" encoding="UTF-8"?>
+  <svg width="50px" height="50px" viewBox="0 0 50 50" version="1.1" xmlns="http://www.w3.org/2000/svg"
+    xmlns:xlink="http://www.w3.org/1999/xlink">
+    <defs>
+      <path id="cggen.\(id)" d="\(pathDescription)"/>
+    </defs>
+    <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" stroke-linejoin="round">
+      <use xlink:href="#cggen.\(id)" stroke="indianred" stroke-width="1"/>
+    </g>
+  </svg>
+  """
+}
+
+extension CGPath {
+  func isAlmostEqual(to other: CGPath, tolerance: CGFloat) -> Bool {
+    let pairs = zip(segments, other.segments)
+    for (lhsElement, rhsElement) in pairs {
+      guard lhsElement.isAlmostEqual(to: rhsElement, tolerance: tolerance) else { return false }
+    }
+    return true
+  }
+
+  private var segments: [PathSegment] {
+    var output = [PathSegment]()
+    applyWithBlock { p in
+      let element = p.pointee
+      output.append(element.pathSegment)
+    }
+    return output
+  }
+}
+
+extension PathSegment {
+  func isAlmostEqual(to other: PathSegment, tolerance: CGFloat) -> Bool {
+    let lhs = self
+    switch (lhs, other) {
+    case let (.moveTo(a), .moveTo(b)):
+      return a.isAlmostEqual(to: b, tolerance: tolerance)
+    case let (.lineTo(a), .lineTo(b)):
+      return a.isAlmostEqual(to: b, tolerance: tolerance)
+    case let (.curveTo(a1, a2, a3), .curveTo(b1, b2, b3)):
+      return a1.isAlmostEqual(to: b1, tolerance: tolerance) &&
+        a2.isAlmostEqual(to: b2, tolerance: tolerance) &&
+        a3.isAlmostEqual(to: b3, tolerance: tolerance)
+    case let (.addArc(a1, a2, a3, a4, a5), .addArc(b1, b2, b3, b4, b5)):
+      return a1.isAlmostEqual(to: b1, tolerance: tolerance) &&
+        a2.isAlmostEqual(to: b2, tolerance: tolerance) &&
+        a3.isAlmostEqual(to: b3, tolerance: tolerance) &&
+        a4.isAlmostEqual(to: b4, tolerance: tolerance) &&
+        a5 == b5
+    case let (.lines(arr1), .lines(arr2)):
+      if arr1.count != arr2.count { return false }
+      let pairs = zip(arr1, arr2)
+      for (p1, p2) in pairs {
+        if !p1.isAlmostEqual(to: p2, tolerance: tolerance) {
+          return false
+        }
+      }
+      return true
+    default: return false
```

Comment:
It's better to list all possible cases here for compiler to notice when new case will be added:
```
case (.moveTo, _), (. lineTo, _), (.curveTo, _) ...:
  return false
```